### PR TITLE
Add rudimentary object model

### PR DESCRIPTION
**Pull request by carlos-zamora**
I added some objects under an "ObjectModel" folder. This is far from ideal. But it's a hack.

Using these objects, you should be able to do XAML Data Binding. AppSettings is the top-level settings object. All of the XAML should be able to retrieve the relevant parts of the AppSettings and update it's innards.

In my other branch, I'm actually hooking up these similar objects to the JSON and Windows Terminal. Once I can get that working, I'll handle moving the Settings UI project over to the main .sln file, then we can actually consume a real `AppSettings.

NOTE: I didn't really define any of the enums. I'm a bit too lazy. You could just populate them as you go with the XAML for now.

### CODE REVIEW

**Comment by carlos-zamora** [comment]
Oh yeah, after I get the other branch done, I think I'll just delete everything in "ObjectModel". That's why I saved everything there for now :P

**Comment by vineeththomasalex** [comment]
So is this building now and can be merged in?

**Comment by carlos-zamora** [comment]
Yup! These are just useless objects for now

File: src/TerminalSettings/TerminalSettings/MainPage.cpp [hunk 1]
```diff
@@ -14,6 +14,11 @@ namespace winrt::TerminalSettings::implementation
     MainPage::MainPage()
     {
         InitializeComponent();
+
+        // TODO: When we actually connect this to Windows Terminal,
```
Comment (vineeththomasalex) [inline src/TerminalSettings/TerminalSettings/MainPage.cpp:22]: You mean read/write the settings json file?

Comment (carlos-zamora) [reply src/TerminalSettings/TerminalSettings/MainPage.cpp:22]: Sorta? Like, WT has some objects in `TerminalSettings` (that's what I'm working on now). I'll replace these "fake" objects with the real ones when they're done :)

**Comment by vineeththomasalex** [comment]
Works locally for me. Can merge.

**Pull request merged**